### ISSUE TITLE
[Feature Request] Add functionality to bake from the command line

### ISSUE THREAD

**Issue by bholzer** [rlguy/Blender-FLIP-Fluids#329]
My rendering/baking pipeline is in the cloud, so much of my baking and rendering happens from the command line on a server, generally with some startup python script.

There is info here on how to bake blender's built-in fluids via Python: https://blender.stackexchange.com/a/728/53213

Could you write documentation on how to do this in Flip?

**Comment by rlguy** [comment]
Hey bholzer,

I didn't have functionality for baking from the command line in mind when writing the baking process so this is not possible within the addon at the moment, but it is a great idea to have this! I have added this functionality and it will be available in the next version (**1.0.4**).

If you want to try it out now, here's how (assuming you are currently using version 1.0.3):

---

You will need to replace the addon ```operators\bake_operators.py``` file with this modified version of the file: [bake_operators.py](https://pastebin.com/FZdPNBs7)

For example, my **bake_operators.py** file is located here:
```
C:\Users\ryanl\AppData\Roaming\Blender Foundation\Blender\2.79\scripts\addons\flip_fluids_addon\operators\bake_operators.py
```

Once the file is replaced, a fluid simulation scene can be baked from the command line using the following script:
```python
import bpy
bpy.ops.flip_fluid_operators.bake_fluid_simulation_cmd()
```

using the same command as you would baking with the built-in fluid simulator:

```
blender --background file.blend --python script.py
```

**Notes:** This script will resume the fluid simulation from the last computed frame in the cache. If you want to reset the simulation cache, you may do so through the Blender scene as usual or by running the following script:

```
import bpy
bpy.ops.flip_fluid_operators.reset_bake()
```

---

I will add this to the documentation after some further testing and closer to the release of the next version.

**Comment by bholzer** [comment]
Awesome, thanks so much, this should make my life easy!

**Comment by bholzer** [comment]
@rlguy I ended up with an error:

```Exporting simulation data...
Error: Traceback (most recent call last):
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/operators/bake_operators.py", line 469, in execute
    self._export_simulation_data(context)
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/operators/bake_operators.py", line 394, in _export_simulation_data
    self._export_simulation_data_file()
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/operators/bake_operators.py", line 363, in _export_simulation_data_file
    dprops.bake.export_filepath
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/export.py", line 432, in export
    __export_simulation_data_to_file(context, simulation_objects, filepath)
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/export.py", line 35, in __export_simulation_data_to_file
    data = __get_simulation_data_dict(context, simobjects)
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/export.py", line 122, in __get_simulation_data_dict
    data['domain_data'] = __get_domain_data_dict(context, simobjects.domain)
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/export.py", line 185, in __get_domain_data_dict
    preferences = bpy.context.user_preferences.addons[id_name].preferences
KeyError: 'bpy_prop_collection[key]: key "flip_fluids_addon" not found'

location: /opt/blender/2.79/scripts/modules/bpy/ops.py:189

Traceback (most recent call last):
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/operators/bake_operators.py", line 469, in execute
    self._export_simulation_data(context)
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/operators/bake_operators.py", line 394, in _export_simulation_data
    self._export_simulation_data_file()
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/operators/bake_operators.py", line 363, in _export_simulation_data_file
    dprops.bake.export_filepath
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/export.py", line 432, in export
    __export_simulation_data_to_file(context, simulation_objects, filepath)
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/export.py", line 35, in __export_simulation_data_to_file
    data = __get_simulation_data_dict(context, simobjects)
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/export.py", line 122, in __get_simulation_data_dict
    data['domain_data'] = __get_domain_data_dict(context, simobjects.domain)
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/export.py", line 185, in __get_domain_data_dict
    preferences = bpy.context.user_preferences.addons[id_name].preferences
KeyError: 'bpy_prop_collection[key]: key "flip_fluids_addon" not found'

location: /opt/blender/2.79/scripts/modules/bpy/ops.py:189

location: /opt/blender/2.79/scripts/modules/bpy/ops.py:189
Error: Traceback (most recent call last):
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/operators/bake_operators.py", line 469, in execute
    self._export_simulation_data(context)
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/operators/bake_operators.py", line 394, in _export_simulation_data
    self._export_simulation_data_file()
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/operators/bake_operators.py", line 363, in _export_simulation_data_file
    dprops.bake.export_filepath
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/export.py", line 432, in export
    __export_simulation_data_to_file(context, simulation_objects, filepath)
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/export.py", line 35, in __export_simulation_data_to_file
    data = __get_simulation_data_dict(context, simobjects)
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/export.py", line 122, in __get_simulation_data_dict
    data['domain_data'] Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/opt/blender/2.79/scripts/modules/bpy/ops.py", line 189, in __call__
    ret = op_call(self.idname_py(), None, kw)
RuntimeError: Error: Traceback (most recent call last):
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/operators/bake_operators.py", line 469, in execute
    self._export_simulation_data(context)
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/operators/bake_operators.py", line 394, in _export_simulation_data
    self._export_simulation_data_file()
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/operators/bake_operators.py", line 363, in _export_simulation_data_file
    dprops.bake.export_filepath
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/export.py", line 432, in export
    __export_simulation_data_to_file(context, simulation_objects, filepath)
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/export.py", line 35, in __export_simulation_data_to_file
    data = __get_simulation_data_dict(context, simobjects)
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/export.py", line 122, in __get_simulation_data_dict
    data['domain_data'] = __get_domain_data_dict(context, simobjects.domain)
  File "/opt/blender/2.79/scripts/addons/flip_fluids_addon/export.py", line 185, in __get_domain_data_dict
    preferences = bpy.context.user_preferences.addons[id_name].preferences
KeyError: 'bpy_prop_collection[key]: key "flip_fluids_addon" not found'

location: /opt/blender/2.79/scripts/modules/bpy/ops.py:189```

**Comment by rlguy** [comment]
Documentation for this feature has been added here: https://github.com/rlguy/Blender-FLIP-Fluids/wiki/Baking-from-the-Command-Line

This feature will be available in the next version update (**1.0.4** available August 11th).
